crates/oxc_parser/src/modifiers.rs

```diff
@@ -178,20 +286,52 @@ impl ModifierKind {
         }
     }
 }
+impl From<Kind> for ModifierKind {
+    // NOTE (@DonIsaac) should we be using TryFrom here? or does unreachable!()
+    // give us compiler optimizations that we want?
+    fn from(kind: Kind) -> Self {
+        match kind {
+            Kind::Abstract => Self::Abstract,
+            Kind::Declare => Self::Declare,
+            Kind::Private => Self::Private,
+            Kind::Protected => Self::Protected,
+            Kind::Public => Self::Public,
+            Kind::Static => Self::Static,
+            Kind::Readonly => Self::Readonly,
+            Kind::Override => Self::Override,
+            Kind::Async => Self::Async,
+            Kind::Const => Self::Const,
+            Kind::In => Self::In,
+            Kind::Out => Self::Out,
+            Kind::Export => Self::Export,
+            Kind::Default => Self::Default,
+            Kind::Accessor => Self::Accessor,
+            _ => unreachable!(),
+        }
+    }
+}
+
+impl std::fmt::Display for ModifierKind {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str(self.as_str())
+    }
+}
 
 impl<'a> ParserImpl<'a> {
-    pub(crate) fn eat_modifiers_before_declaration(&mut self) -> (ModifierFlags, Modifiers<'a>) {
+    pub(crate) fn eat_modifiers_before_declaration(&mut self) -> Modifiers<'a> {
         let mut flags = ModifierFlags::empty();
         let mut modifiers = self.ast.new_vec();
         while self.at_modifier() {
             let span = self.start_span();
             let modifier_flag = self.cur_kind().into();
-            flags.set(modifier_flag, true);
             let kind = self.cur_kind();
             self.bump_any();
-            modifiers.push(Self::modifier(kind, self.end_span(span)));
+            let modifier = Self::modifier(kind, self.end_span(span));
+            self.verify_modifier(flags, &modifier);
```

Comment:
I'd rather do this in the checker, but we don't persist modifiers in the AST and so we lose spans